google_guest_agent/diagnostics.go

```diff
@@ -116,13 +116,13 @@ func (d *diagnosticsMgr) set(ctx context.Context) error {
 		logger.Infof("Diagnostics: reject the request, as an existing process is collecting logs from the system")
 		return nil
 	}
-	cmd := exec.Command(diagnosticsCmd, args...)
+
 	go func() {
 		logger.Infof("Diagnostics: collecting logs from the system.")
-		out, err := cmd.CombinedOutput()
-		logger.Infof(string(out[:]))
-		if err != nil {
-			logger.Infof("Error collecting logs: %v", err)
+		res := run.WithCombinedOutput(ctx, diagnosticsCmd, args...)
+		logger.Infof(res.Combined)
+		if res.ExitCode != 0 {
+			logger.Infof("Error collecting logs: %v", err.Error())
```

Comment:
sorry missed this while reviewing earlier. I believe instead of `err.Error()` this should be `res. StdErr` now? There's no `err` in this go routine now.

Also, I see it was logged at `info` level before but considering its an error do we want to change it to `warn`?

Reply:
Good catch, it should be `res.Error()` actually. Sure I'll change the logging to warn.

Reply:
done. Thanks.